pymemcache/client/hash.py

```diff
@@ -0,0 +1,282 @@
+import socket
+import time
+from pymemcache.client.base import Client, PooledClient, _check_key
+from clandestined import RendezvousHash as RH
+
+
+class RendezvousHash(RH):
+    def get_node(self, key):
+        return self.find_node(key)
+
+
+class HashClient(object):
+    """
+    A client for communicating with a cluster of memcached servers
+    """
+    def __init__(
+        self,
+        servers,
+        hasher=None,
+        serializer=None,
+        connect_timeout=None,
+        timeout=None,
+        no_delay=False,
+        socket_module=socket,
+        key_prefix=b'',
+        max_pool_size=None,
+        lock_generator=None,
+        retry_attempts=2,
+        retry_timeout=1,
+        dead_timeout=60,
+        use_pooling=False,
+    ):
+        """
+        Args:
+          servers: list(tuple(hostname, port))
+          serializer: optional class with ``serialize`` and ``deserialize``
```

Comment:
Yeah, I was trying to improve things for the future but I think instead of changing the contract during this PR I'll introduce a `v2` directory that changes the construct and raises deprecation warnings on the others in a different PR